library/core/src/str/pattern.rs

```diff
@@ -1965,3 +1977,45 @@ unsafe fn small_slice_eq(x: &[u8], y: &[u8]) -> bool {
         vx == vy
     }
 }
+
+#[cfg(kani)]
+#[unstable(feature = "kani", issue = "none")]
+pub mod verify {
+    use super::*;
+
+    #[cfg(all(kani, target_arch = "x86_64"))] // only called on x86
+    #[kani::proof]
+    #[kani::unwind(4)]
```

Comment:
Why do we still need bounds in this case?

Reply:
There is another loop in `small_slice_eq` for the case that the size of slice is less than 4.